Some names in .ecore file can cause compilation errors

Some names used in a source .ecore file cause compilation errors after following the basic tutorial.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<ecore:EPackage xmi:version="2.0" xmlns:xmi="http://www.omg.org/XMI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="model" nsURI="http://ns.ansys.com/test_model" nsPrefix="model">
  <eClassifiers xsi:type="ecore:EClass" name="Element" abstract="true">
    <eStructuralFeatures xsi:type="ecore:EAttribute" name="name" lowerBound="1" eType="ecore:EDataType http://www.eclipse.org/emf/2002/Ecore#//EString"
        iD="true"/>
  </eClassifiers>
  <eClassifiers xsi:type="ecore:EClass" name="Tree" abstract="true" eSuperTypes="#//Element">
    <eStructuralFeatures xsi:type="ecore:EReference" name="children" upperBound="-1"
        eType="#//Element" containment="true"/>
  </eClassifiers>
</ecore:EPackage>
```

Compilation errors:
```
1>------ Build started: Project: NMFDemo, Configuration: Debug Any CPU ------
1>C:\Users\nsharp\source\repos\NMFDemo\NMFDemo\Metamodel\ITree.cs(53,41,53,49): warning CS0108: 'ITree.Children' hides inherited member 'IModelElement.Children'. Use the new keyword if hiding was intended.
1>C:\Users\nsharp\source\repos\NMFDemo\NMFDemo\Metamodel\Element.cs(226,51,226,56): error CS0118: 'Model' is a namespace but is used like a type
1>C:\Users\nsharp\source\repos\NMFDemo\NMFDemo\Metamodel\Element.cs(226,67,226,72): error CS0118: 'Model' is a namespace but is used like a type
1>C:\Users\nsharp\source\repos\NMFDemo\NMFDemo\Metamodel\Tree.cs(71,48,71,56): warning CS0114: 'Tree.Children' hides inherited member 'ModelElement.Children'. To make the current member override that implementation, add the override keyword. Otherwise add the new keyword.
1>C:\Users\nsharp\source\repos\NMFDemo\NMFDemo\Metamodel\Tree.cs(82,62,82,70): error CS0102: The type 'Tree' already contains a definition for 'Children'
1>Done building project "NMFDemo.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

I am using VS 2019 with NMF-Basic 2.0.176.

Renaming 'model' to 'model2' and 'children' to 'children2' resolves the issue, but changes the DSL.

--- a/NMF.sln
+++ b/NMF.sln
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.29424.173
+# Visual Studio Version 17
+VisualStudioVersion = 17.0.31919.166
 MinimumVisualStudioVersion = 10.0.40219.1
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Analysis", "Analysis", "{DB96042F-8815-430D-8345-362F4266F0FE}"
 EndProject

--- a/Transformations/Models.MetaTransformation/Meta/Reference2Property.cs
+++ b/Transformations/Models.MetaTransformation/Meta/Reference2Property.cs
@@ -40,12 +40,25 @@ namespace NMF.Models.Meta
                 Call(Rule<Type2Type>(), reference => reference.Type);
             }
 
+            private static string[] ReservedPropertyNames = new string[]
+            {
+                nameof(ModelElement.Children),
+                nameof(ModelElement.ReferencedElements),
+                nameof(ModelElement.IsFrozen),
+                nameof(ModelElement.IsIdentified),
+                nameof(ModelElement.IsLocked),
+                nameof(ModelElement.Model),
+                nameof(ModelElement.Parent),
+                nameof(ModelElement.RelativeUri),
+                nameof(ModelElement.AbsoluteUri)
+            };
+
             /// <inheritdoc />
             public override CodeMemberProperty CreateOutput(IReference input, ITransformationContext context)
             {
                 var property = new CodeMemberProperty();
                 property.Name = input.Name.ToPascalCase();
-                if (property.Name == input.DeclaringType.Name.ToPascalCase())
+                if (property.Name == input.DeclaringType.Name.ToPascalCase() || ReservedPropertyNames.Contains(property.Name))
                 {
                     property.Name += "_";
                 }

--- a/Transformations/Tests/CodeGenerationTests/CodeGenerationTests.csproj
+++ b/Transformations/Tests/CodeGenerationTests/CodeGenerationTests.csproj
@@ -62,6 +62,9 @@
         <None Update="KDM.ecore">
             <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
         </None>
+        <None Update="NameClashes.ecore">
+          <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+        </None>
         <None Update="Persons.ecore">
             <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
         </None>

--- a/Transformations/Tests/CodeGenerationTests/ModelTests.cs
+++ b/Transformations/Tests/CodeGenerationTests/ModelTests.cs
@@ -76,6 +76,12 @@ namespace NMF.CodeGenerationTests
         }
 
         [TestMethod]
+        public void NameClashesResolvedSuccessfully()
+        {
+            GenerateAndAssertEcore("NameClashes.ecore");
+        }
+
+        [TestMethod]
         public void DefaultValueExampleGeneratesAndInstanceCanBeLoaded()
         {
             GenerateAndAssertEcore("DefaultValueTest.ecore");

--- a/Transformations/Tests/CodeGenerationTests/NameClashes.ecore
+++ b/Transformations/Tests/CodeGenerationTests/NameClashes.ecore
@@ -1,0 +1,12 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
+<ecore:EPackage xmi:version="2.0" xmlns:xmi="http://www.omg.org/XMI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore" name="nameClashes" nsURI="http://example.com/nameClashes" nsPrefix="model">
+	<eClassifiers xsi:type="ecore:EClass" name="Element" abstract="true">
+		<eStructuralFeatures xsi:type="ecore:EAttribute" name="name" lowerBound="1" eType="ecore:EDataType http://www.eclipse.org/emf/2002/Ecore#//EString"
+			iD="true"/>
+	</eClassifiers>
+	<eClassifiers xsi:type="ecore:EClass" name="Tree" abstract="true" eSuperTypes="#//Element">
+		<eStructuralFeatures xsi:type="ecore:EReference" name="children" upperBound="-1"
+			eType="#//Element" containment="true"/>
+	</eClassifiers>
+</ecore:EPackage>

